pxr/usdImaging/usdviewq/appController.py

```diff
@@ -2172,29 +2175,66 @@ def setFrameField(self, frame):
 
     # Prim/Attribute search functionality =====================================
 
+    def _isMatch(self, pattern, isRegex, prim, useDisplayName):
+        """
+        Determines if the given prim has a name that matches the
+        given pattern.  If useDisplayName is True, the match
+        will be performed on the prim's display name (if authored)
+        and on the prim's name (if not).  When useDisplayName is False,
+        the match is always performed against the prim's name.
+
+        Args:
+            pattern (str): The pattern to use to match the name.  Pattern
+                           is either a sequence of characters or a regex
+                           expression.  If it is a regex expression, the
+                           isRegex parameter should be set to True.
+            isRegex (bool): True if the given pattern is a regex expression
+                            or False if just a sequence of characters.
+            prim (object): A python facing UsdPrim object on whos properties
+                           should be matched by pattern.
+            useDisplayName (bool): True if the pattern match should be against
+                                   the displayName of the prim or False if
+                                   against the name of the prim.  If this value is True
+                                   displayName will only be matched if it is authored,
+                                   otherwise the name of the prim will be used.
+
+        Returns:
+            True if the pattern matches the specified prim content, False otherwise. 
+        """
+        if isRegex:
+            matchLambda = re.compile(pattern, re.IGNORECASE).search
+        else:
+            pattern = pattern.lower()
+            matchLambda = lambda x: pattern in x.lower()
+
+        if useDisplayName:
+            if prim.HasAuthoredDisplayName():
+                return matchLambda(prim.GetDisplayName())
+            else:
+                return matchLambda(prim.GetName())
```

Comment:
Seeing the logic so nicely laid out here reminds me that to avoid angering the python performance gods on this set of tests that could run millions of times in many scenes (since we usually process all instance proxies, as well!), we'd likely be considerably better off _just_ calling `GetDisplayName()` up front and comparing switching on a non-empty result, rather than making the extra `HasAuthoredDisplayName()` API call (which, internally, does _just_ about the same amount of work as `GetDisplayName()`, incidentally).  If you use the `--timing` usdview option, I believe that each search will print out how long it takes, so if you're curious, you could do that profiling.  I might be worrying needlessly here, but this kind of thing is exactly why we created the PrimInfo structure/dance to service PrimViewItems, to substantial benefit.

Reply:
You are correct - running `usdview` with the `--timing` option does show a slight decrease in search speed if we get the display name and compare to `None` / `""` vs. checking for authored first.  These are small scenes, but this slight difference will add up in larger ones.  Made the change.